src/main/java/com/pagsestagio/movieapi/kafka/FilmePesquisadoConsumer.java

```diff
@@ -0,0 +1,61 @@
+package com.pagsestagio.movieapi.kafka;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.pagsestagio.movieapi.model.EventoFilmePesquisado;
+import com.pagsestagio.movieapi.model.FilmeEstatistica;
+import com.pagsestagio.movieapi.repository.FilmeEstatisticaRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.support.Acknowledgment;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class FilmePesquisadoConsumer {
+
+    private final FilmeEstatisticaRepository filmeEstatisticaRepository;
+
+    private final ObjectMapper objectMapper;
+    private static final Logger logger = LoggerFactory.getLogger(FilmePesquisadoConsumer.class);
+
+    public FilmePesquisadoConsumer(FilmeEstatisticaRepository filmeEstatisticaRepository, ObjectMapper objectMapper) {
+        this.filmeEstatisticaRepository = filmeEstatisticaRepository;
+        this.objectMapper = objectMapper;
+    }
+
+    @KafkaListener(topics = "movie-api.filme-pesquisado", groupId = "filme-estatistica-group")
+    public void consomeMensagem(@Payload String mensagem, Acknowledgment ack) throws JsonProcessingException {
```

Comment:
qual é a key desse evento?

lembra que os eventos são apenas ordenados por partição, e a partição é definida pela chave do evento. 
eventos da mesma chave caem na mesma partição em ordem.

Por que isso é relevante? 
Se você tiver eventos para o mesmo filme indo para consumidores diferentes isso geraria concorrencia na atualização do registro do banco, o que levaria a inconsistencia. 

tipo, acontece ao mesmo tempo, 10 vezes a atualização do contador. o que não final acaba que só tenha só mais um no contador. 
(todas a instancias pegaram o registro com contagem x,  e ae atualizaram para x+1; como isso ocorre ao mesmo tempo, as alterações que outras instancias estão fazendo são invisíveis e todas atualizaram com só mais um)

para evitar isso ou você garante ordenação (que o filme sempre caia um por um no msmo consumidor)
ou precisa dar um lock no registro no banco

Reply:
![image](https://github.com/user-attachments/assets/5e164d3b-ae88-43a7-b184-7beabe685d60)
Você fala essa key aqui? Se for é o idPublico do filme, o UUID.